src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -248,36 +250,59 @@ def validate_paths(self, data, schema_name, verrors):
             verrors.add(f"{schema_name}.alldirs", "This option can only be used for datasets")
 
     @private
-    def validate_user_networks(self, other_shares, data, schema_name, verrors):
+    def validate_hosts_and_networks(self, other_shares, data, schema_name, verrors):
         dev = os.stat(data["paths"][0]).st_dev
 
-        used_networks = []
+        used_networks = defaultdict(lambda: 0)
         for share in other_shares:
             try:
                 share_dev = os.stat(share["paths"][0]).st_dev
             except Exception:
                 self.logger.warning("Failed to stat first path for %r", share, exc_info=True)
                 continue
 
-            used_networks.extend([(network, share_dev) for network in share["networks"]])
-
-            if data["alldirs"] and share["alldirs"] and share_dev == dev:
-                verrors.add(f"{schema_name}.alldirs", "This option is only available once per mountpoint")
+            if share_dev == dev:
+                for host in share["hosts"]:
+                    try:
+                        network = ipaddress.ip_network(f"{host}/32")
```

Comment:
Shouldnt we allow hostnames here again?

Reply:
I think no because they corresponding IP addresses can change and we won't be able to perform our checks

Reply:
Changing is actually a feature.

And which checks? Hostnames are always /32, it should be possible to always share that, onless its repeated, then it doesnt matter. Or am I missing something?

Reply:
Changing is a bad feature. NFS resolves hostnames at start. Then they may change but it won't know. We can resolve hostnames here, we can resolve them on config file generation but if they can change, we can't guarantee that we always generate a valid config.

Reply:
We have allowed hostnames for years. What do we tell and do with people using them?

Reply:
Sorry, I've missed the fact that hostnames were allowed in the past.